src/client/components/Search/useSearch.tsx

```diff
@@ -0,0 +1,292 @@
+import { ApiSuccessResponse, isError } from '../../../universal/helpers/api';
+import { AppState } from '../../AppState';
+import { addAxiosResponseTransform } from '../../hooks/api/useDataApi';
+import { appStateAtom, useAppStateGetter } from '../../hooks/useAppState';
+import {
+  profileTypeState,
+  useProfileTypeSwitch,
+  useProfileTypeValue,
+} from '../../hooks/useProfileType';
+import {
+  ApiBaseItem,
+  ApiSearchConfig,
+  displayPath,
+  getApiSearchConfigs,
+  SearchEntry,
+  staticIndex,
+} from './searchConfig';
+
+export function generateSearchIndexPageEntry(
+  item: ApiBaseItem,
+  apiConfig: ApiSearchConfig
+): SearchEntry {
+  const props: Array<
+    Exclude<keyof ApiSearchConfig, 'getApiBaseItems' | 'apiName'>
+  > = [
+    'keywordSourceProps',
+    'title',
+    'displayTitle',
+    'description',
+    'url',
+    'keywords',
+  ];
+
+  const pageEntry = {} as SearchEntry;
+
+  for (const prop of props) {
+    const configValue = apiConfig[prop];
+
+    let value =
+      typeof configValue === 'function'
+        ? configValue(item, apiConfig)
+        : configValue;
+
+    if (!value) {
+      continue;
+    }
+
+    if (prop === 'keywords') {
+      pageEntry.keywords = uniqueArray([
+        ...(pageEntry.keywords || []),
+        ...value,
+      ]);
+    } else if (prop === 'keywordSourceProps') {
+      value = Object.values(pick(item, value));
+      pageEntry.keywords = uniqueArray([
+        ...(pageEntry.keywords || []),
+        ...value,
+      ]);
+    } else {
+      pageEntry[prop as keyof SearchEntry] = value;
+    }
+  }
+
+  return pageEntry;
+}
+
+export function generateSearchIndexPageEntries(
+  profileType: ProfileType,
+  apiName: string,
+  apiContent: ApiSuccessResponse<any>['content']
+): SearchEntry[] {
+  const apiConfig = getApiSearchConfigs(profileType).find(
+    (config) => config.apiName === apiName
+  );
+
+  if (!apiConfig) {
+    throw new Error(`${apiName} does not have search index entry.`);
+  }
+
+  return apiConfig
+    .getApiBaseItems(apiContent)
+    .map((item) => generateSearchIndexPageEntry(item, apiConfig));
+}
+
+interface AmsterdamSearchResult {
+  title: string;
+  sections: string[];
+  description: string;
+  url: string;
+  highlight: { title: string };
+}
+
+function transformSearchAmsterdamNLresponse(responseData: any): SearchEntry[] {
+  if (Array.isArray(responseData?.records?.page)) {
+    return responseData.records.page.map((page: AmsterdamSearchResult) => {
+      return {
+        title: page.title,
+        displayTitle: (term: string) =>
+          displayPath(term, [page.highlight.title || page.title], true, false),
+        keywords: page.sections,
+        description: page.description,
+        url: page.url,
+      };
+    });
+  }
+  return [];
+}
+
+export async function searchAmsterdamNL(
+  keywords: string,
+  resultCountPerPage: number = 5,
+  useExtendedAmsterdamSearch: boolean = false
+) {
+  const url = useExtendedAmsterdamSearch
```

Comment:
use is een lastige term geworden sinds hooks. Mijn voorkeur heeft het use dan ook alleen maar voor hooks te gebruiken.